tests/test_unit/test_load_poses.py

```diff
@@ -2,13 +2,32 @@
 import pytest
 import xarray as xr
 from pytest import POSE_DATA
+from sleap_io.io.slp import read_labels
 
 from movement.io import PosesAccessor, load_poses
 
 
 class TestLoadPoses:
     """Test suite for the load_poses module."""
 
+    @pytest.fixture(scope="module")
+    def sleap_user_labels(self):
+        """Return a SLEAP `Labels` object containing only
+        user-labeled instances.
+        """
+        slp_file = POSE_DATA.get(
+            "SLEAP_three-mice_Aeon_proofread.predictions.slp"
+        )
+        return read_labels(slp_file)
+
+    @pytest.fixture(scope="module")
+    def sleap_predicted_labels(self):
+        """Return a SLEAP `Labels` object containing only
+        predicted instances.
+        """
+        slp_file = POSE_DATA.get("SLEAP_single-mouse_EPM.predictions.slp")
+        return read_labels(slp_file)
```

Comment:
I think this concerns a separate issue. For simplicity, let's consider a single-animal frame here.
Possible scenarios leading to a mix of Predicted and User-labelled Instances in a `sleap.instance.LabeledFrame`:
- case 1 - user "fixes" incorrectly predicted individual A: 
  `[PredictedInstance(A)]` &rarr; `[PredictedInstance(A), Instance(A)]` 
- case 2 - user "adds" missing individual A: 
  `[]` &rarr; `[Instance(A)]`  

In either case, the current behaviour using `sleap-io`'s [`Labels.numpy()`](https://github.com/talmolab/sleap-io/blob/9a04f7feac0397d7d7bed9fb79f205678cccf9f8/sleap_io/model/labels.py#L92) is to ignore user-labelled instances:
- loading `case1.slp` in _movement_ returns `[PredictedInstance(A)]`
- loading `case2.slp` in _movement_ returns `[]`

However, if the user exports `case1.slp` as `case1.h5`, and `case2.slp` as `case2.h5`, SLEAP's [`write_tracking_h5` prioritises the user-labelled instances](https://github.com/talmolab/sleap/blob/9eeba979b7ce32400809723b668a8bd41e09d973/sleap/info/write_tracking_h5.py#L138-L141), i.e., if user-labelled instance is present, select user-labelled instance, otherwise select predicted instance. Hence:
- loading `case1.h5` in _movement_ returns `[Instance(A)]` with NaN confidence
- loading `case2.h5` in _movement_ returns `[Instance(A)]` with NaN confidence

I guess we should handle .slp files with mixed instance types the same way SLEAP does in `write_tracking_h5`, so that loading `.slp` and `.h5`  in _movement_ return the same dataset. And perhaps, for consistency, we should leave confidence scores for user-labels-only files as NaNs, rather than setting these as 1.0... 

Wrt test data, I think it's more straightforward to add a pair of .h5 and .slp files to our sample datasets.

Wdyt?

Reply:
I thought about it, and I think what you propose here makes sense and aligns better with the expectations of users. If I correct a prediction (thereby, turning it into a user-labelled instance), I expect it to be prioritised for going into the output, like `write_tracking_h5` does.

Moreover, as you point out, it makes sense for the imported data to be the same, whether it came from `.slp` or `.h5`. This means that we should indeed duplicate the functionality of `write_tracking_h5`. Since we cannot depend on `sleap`, and `sleap-io` doesn't have it, we should write our own implementation for it.

There are two issues I foresee with this approach:
1. We have to keep an eye on the `write_tracking_h5` in `sleap`. If it's behaviour changes significantly, we may need to update our own version. But I hope they won't do that, because if they do, we'll have to ship different data loaders for different versions of `sleap`.  But I guess this is the case already, we have to stay up-to-date with developments in `deeplabcut` and `sleap`.
2. As you mention, we probably have to mark the user-labelled instances with NaN confidence. This will not be a problem if we assume that user-labelling is the only process that can generate NaNs in `.slp` and `.h5` files. If this assumption holds, we can treat NaNs coming from `sleap` as having the highest confidence during preprocessing. If this assumption is not true, we need another way of keeping track of which instances are user-labelled (maybe an additional boolean `DataArray` or a list of the indices of  user-labelled instances stored in the `Dataset`'s metadata `attrs`.

In summary, I think you should go ahead with implementing this plan in this PR, and we should come to a decision on how to handle NaNs. We definitely need to keep information on which predictions we can "trust",  because it will come in handy during data cleaning.